### PR TITLE
_scripts: build docker images using --secret

### PR DESCRIPTION
**Pull request by myitcv**
This ensures that we don't expose build args as part of the resulting
image.

### CODE REVIEW

File: _scripts/Dockerfile.gvim [hunk 1]
```diff
@@ -24,7 +25,8 @@ ENV PATH=/vim/bin:$PATH
 RUN rm ~/.netrc
 
 ARG GO_VERSION
-RUN curl -sL https://dl.google.com/go/${GO_VERSION}.linux-amd64.tar.gz | tar -C / -zx
+ARG TARGETARCH TARGETOS
```
Comment (leitzler) [inline _scripts/Dockerfile.gvim:28]: Leftovers from multi-arch experiments? Can't see that they are set anywhere.

File: _scripts/Dockerfile.vim [hunk 2]
```diff
@@ -24,7 +25,8 @@ ENV PATH=/vim/bin:$PATH
 RUN rm ~/.netrc
 
 ARG GO_VERSION
-RUN curl -sL https://dl.google.com/go/${GO_VERSION}.linux-amd64.tar.gz | tar -C / -zx
+ARG TARGETARCH TARGETOS
```
Comment (leitzler) [inline _scripts/Dockerfile.vim:28]: Same

**Review by leitzler — APPROVED** [review]
Just a comment, otherwise LGTM

File: _scripts/Dockerfile.gvim (hunk 1 above)
Comment (myitcv) [reply _scripts/Dockerfile.gvim:28]: Exactly.

Comment (myitcv) [reply _scripts/Dockerfile.gvim:28]: Actually, just to check. Are you referring to `TARGET(OS|ARCH)` here?

If so, they are used on the following line.

Or did I misunderstand?

Comment (leitzler) [reply _scripts/Dockerfile.gvim:28]: Yes, as per our Slack conversation I didn't know that they were set automatically by buildkit. All good!

Comment (myitcv) [reply _scripts/Dockerfile.gvim:28]: Thanks for clarifying!

**Pull request merged**